server/routerlicious/packages/lambdas/src/scribe/lambda.ts

```diff
@@ -345,12 +367,38 @@ export class ScribeLambda implements IPartitionLambda {
             }
         }
 
-        const checkpoint = this.generateCheckpoint(message.offset);
-        this.checkpointCore(
-            checkpoint,
-            message,
-            this.clearCache);
-        this.lastOffset = message.offset;
+        // Create the checkpoint
+        this.updateCheckpointMessages(message);
+        this.checkpointInfo.rawMessagesSinceCheckpoint++;
+
+        if (this.noActiveClients) {
+            this.prepareCheckpoint(message, ScribeCheckpointReason.NoClients);
+            this.noActiveClients = false;
+        } else {
+            const checkpointReason = this.getCheckpointReason();
+            if (checkpointReason !== undefined) {
+                // checkpoint the current up-to-date state
+                this.prepareCheckpoint(message, checkpointReason);
+            } else {
+                this.updateCheckpointIdleTimer();
+            }
+        }
+    }
+
+    public prepareCheckpoint(message: IQueuedMessage, checkpointReason: ScribeCheckpointReason) {
+        // Get checkpoint context
+        const checkpoint = this.generateScribeCheckpoint(message.offset);
+        this.updateCheckpointMessages(message);
+
+        // write the checkpoint with the current up-to-date state
+            this.checkpoint(checkpointReason);
```

Comment:
nit: this block seems to be padded with one additional tab that might not be necessary

Reply:
Removed the extra tab